### PR TITLE
add tests and run tests on pull requests fix #22

### LINKED ISSUE #22: add tests
**Describe the bug**
The repo is missing tests. Why a repo is should have tests is a given

**To Reproduce**

**Expected behavior**

Have tests ran in pull requests

**Screenshots**

**version (please complete the following information):**

**Additional context**

See https://github.com/phaazon/hop.nvim/pull/271 for a first approach. Still dunno why was it closed.

### PR DESCRIPTION
**Pull request by FelipeLema**
fixes #22

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    name: unit tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            url: https://github.com/neovim/neovim/releases/download/nightly/nvim-linux64.tar.gz
+          - os: ubuntu-20.04
+            url: https://github.com/neovim/neovim/releases/download/v0.7.0/nvim-linux64.tar.gz
```
Comment (smoka7) [inline .github/workflows/test.yml:16]:
```suggestion
            url: https://github.com/neovim/neovim/releases/download/v0.9.1/nvim-linux64.tar.gz
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  unit_tests:
+    name: unit tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
```
Comment (smoka7) [inline .github/workflows/test.yml:13]:
```suggestion
          - os: ubuntu-latest
```

**Comment by smoka7** [comment]
seems like `override_getchar()` doesn't override it, any idea?

**Comment by FelipeLema** [comment]
we're using both `getchar()` & `getcharstr()`, so I need to override both

should we switch to just using `getcharstr()`? we can also avoid lines converting the code

**Comment by smoka7** [comment]
Thanks I fixed those. If this looks good to you let's merge it.

**Comment by FelipeLema** [comment]
not fully convinced about the "special key" is to be managed, but go ahead and merge... we can take care of it later

**Comment by smoka7** [comment]
Ok, ping me about how you want to approach adding tests.

**Comment by FelipeLema** [comment]
@smoka7 I believe most PRs should be accompanied by tests: one should add a feature/bugfix _and_ tests for it

I'll subscribe to PRs in this repo and help out adding tests to them

**Pull request merged**